yt/data_objects/construction_data_containers.py

```diff
@@ -673,14 +675,28 @@ def get_data(self, fields = None):
                     "with nonzero num_ghost_zones." % self._num_ghost_zones)
             else:
                 raise
-        if len(part) > 0: self._fill_particles(part)
+
+        is_sph_field = False
+        if len(part) > 0:
+            # checking if we have a sph particles
+            is_sph_field = 'gas' in part[0]
+            if(hasattr(self.ds, '_sph_ptype')):
+                is_sph_field = self.ds._sph_ptype in part[0]
+
+            if(is_sph_field):
+                self._fill_sph_particles(fields)
+            else:
+                self._fill_particles(part)
```

Comment:
Let's work through this together during our call tomorrow, I'm a little confused about your concern in the PR description and I think it might be easier to chat about it over a telecon.